src/services/users.ts

```diff
@@ -60,14 +69,39 @@ export class UserService {
     await changePassword(username, newPasswordHash);
   }
 
-  public async changeRole(userId: number, roleType: RoleType, groupId: number): Promise<void> {
-    const membership = await getMembershipById(userId, groupId);
-    if (!membership) {
+  public async changeRole(user: User, userId: number, roleType: RoleType): Promise<void> {
```

Comment:
Тут желательно пометить currentUser и targetUserId. Потому что иначе приходится ломать голову для какого юзера мы меняем роль. Также можно поставить targetUserId и roleType первыми аргументами, а currentUser последним, чтобы семантически более важные аргументы стояли первыми. Можно еще даже добавлять документацию в таких случаях.

```suggestion
/**
* Change role of a user
 * @param targetUserId - User ID of entity that change role.
 * @param roleType - Type of role that are going to be applied to user.
 * @param currentUser - User that is performing change role action.
 */
  public async changeRole(targetUserId: number, roleType: RoleType, currentUser: User): Promise<void> {
```

Не уверен, что я правильно написал suggestion, но возможно сработает :)